pkgs/top-level/all-packages.nix

```diff
@@ -14743,15 +14743,20 @@ with pkgs;
   julia-stable = julia_18;
   julia = julia-stable;
 
-  julia_16-bin = callPackage ../development/compilers/julia/1.6-bin.nix { };
-  julia_18-bin = callPackage ../development/compilers/julia/1.8-bin.nix { };
+  julia_16-bin =  let julia_16-bin' = callPackage ../development/compilers/julia/1.6-bin.nix { };
+                  in callPackage ../development/compilers/julia/with-packages.nix { julia = julia_16-bin'; };
+
+  julia_18-bin = let julia_18-bin' = callPackage ../development/compilers/julia/1.8-bin.nix { };
+                 in callPackage ../development/compilers/julia/with-packages.nix { julia = julia_18-bin'; };
 
   julia_18 = callPackage ../development/compilers/julia/1.8.nix { };
 
   julia-lts-bin = julia_16-bin;
   julia-stable-bin = julia_18-bin;
   julia-bin = julia-stable-bin;
 
+  julia2nix = callPackage ../development/julia-modules/julia2nix.nix { julia = julia-bin; };
+
```

Comment:
The one build from source is equally usable as the binary one, AFAIK. And generally we should prefer source builds.

Reply:
I agree, but I'm getting  `ERROR: Unable to find compatible target in system image.`